app/presenters/vacancies_presenter.rb

```diff
@@ -26,30 +26,59 @@ def any?
     decorated_collection.count.nonzero?
   end
 
-  def total_count_message(keyword)
-    if total_count == 1
-      return I18n.t('jobs.job_count_without_search', count: total_count) unless @searched
+  def search_heading(keyword: '', location: '')
+      case true
+      when (keyword && keyword.length > 0) && (location && location.length > 0)
```

Comment:
Alternatively, `when (keyword.length > 0)` (and same for location).
If keyword is nil, `>` throws an error, but keyword is never not a string.

Reply:
yes this was the main bit of ruby i thought was a bit inelegant, i am used to `def search_heading(keyword: '', location: '')` type declarations meaning that keyword or location are always empty string even if not passed as arguments. and i think that correct as we are dealing with string manipulation/construction here. but i dont see why you would need this part if that was the case `when (keyword && `

Reply:
as in keyword has to exist

Reply:
@cesidio ?

Reply:
In ruby you can use the shorter syntax:

keyword.present? && location.present?`

same for the ones below

Reply:
works like a charm and a lot neater/readable